scylla/src/transport/load_balancing/default.rs

```diff
@@ -1313,6 +1313,101 @@ mod tests {
                     .group([B, C, E]) // remote nodes
                     .build(),
             },
+            // Keyspace SS with RF=2 with enabled DC failover and rack-awareness
+            Test {
+                policy: DefaultPolicy {
+                    preferred_datacenter: Some("eu".to_owned()),
+                    preferred_rack: Some("r2".to_owned()),
+                    is_token_aware: true,
+                    permit_dc_failover: true,
+                    ..Default::default()
+                },
+                routing_info: RoutingInfo {
+                    token: Some(Token { value: 760 }),
+                    keyspace: Some(KEYSPACE_SS_RF_2),
+                    consistency: Consistency::One,
+                    ..Default::default()
+                },
+                // going though the ring, we get order: G , B , A , E , F , C , D
+                //                                      eu  eu  eu  us  us  eu  us
+                //                                      r2  r1  r1  r1  r2  r1  r1
+                expected_groups: ExpectedGroupsBuilder::new()
+                    .group([G]) // pick local rack replicas
+                    .group([B]) // local DC replicas
+                    .group([A, C]) // local nodes
+                    .group([F, D, E]) // remote nodes
+                    .build(),
+            },
+            // Keyspace SS with RF=2 with enabled rack-awareness, shuffling replicas disabled
+            Test {
+                policy: DefaultPolicy {
+                    preferred_datacenter: Some("eu".to_owned()),
+                    preferred_rack: Some("r1".to_owned()),
+                    is_token_aware: true,
+                    permit_dc_failover: false,
+                    fixed_shuffle_seed: Some(123),
+                    ..Default::default()
+                },
+                routing_info: RoutingInfo {
+                    token: Some(Token { value: 560 }),
+                    keyspace: Some(KEYSPACE_SS_RF_2),
+                    consistency: Consistency::Two,
+                    ..Default::default()
+                },
+                // going though the ring, we get order: B , C , E , G , A , F , D
+                //                                      eu  eu  us  eu  eu  us  us
+                //                                      r1  r1  r1  r2  r1  r2  r1
+                expected_groups: ExpectedGroupsBuilder::new()
+                    .deterministic([B, C]) // pick local rack + fallback replicas
```

Comment:
Nit: instead of this single `.deterministic`, you should do:

```rust
.deterministic([B])
.deterministic([C])
```

The `.deterministic()` method does not check that the load balancing policy always returns in the specified order - e.g. in your case it does not check that it always return `B` and then `C`. It only checks that the nodes within the group are always returned in an _unspecified_, but _always the same order_.

In your case, both `[B, C]` and `[C, B]` orders will be accepted by the test as long as the load balancing policy always sticks to the same order. I assume that you wanted to check that `B` then `C` is always returned - in such case you should use two groups like I suggested.